### DISCUSSION TITLE
ObservableValidator() and AOT

### DISCUSSION THREAD

**Discussion by nvgogh** [CommunityToolkit/dotnet#1001] [General]
Looks like the OberservableValidator can't be used with AOT enabled. I get the linker warning when compiling my Maui app with Net9

warning IL2026: Using member 'CommunityToolkit.Mvvm.ComponentModel.ObservableValidator.ObservableValidator()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. The type of the current instance cannot be statically discovered.

What is the best way to work around this?

I searched everywhere but I haven't stumbled across a good hint.

**Comment by Symbai** [comment]
Still doesn't work. Just came here after figuring out that the properties get trimmed out and I receive exceptions like "the type: myviewmodel does not contain a public property named 'myobservablepropertywithvalidation'"
